src/server/game/Entities/Player/Player.cpp

```diff
@@ -3420,6 +3422,9 @@ bool Player::AddSpell(uint32 spellId, bool active, bool learning, bool dependent
         return false;
     }
 
+    if (sDB2Manager.GetMount(spellId))
```

Comment:
If I remove the one in AddSpell all spells will go into AddMount which just looks bad.
If I remove the one in AddMount I'm risking that a spell which used to be mount but now isn't is loaded as mount (and it's general prevention for data corruption inside that table).